features/step_definitions/reek_steps.rb

```diff
@@ -11,69 +11,72 @@
 end
 
 Then /^it reports nothing$/ do
-  assert_exact_output '', all_stdout
+  expect(last_command_started).to have_output_on_stdout('')
 end
 
 Then /^there is no output on stdout$/ do
-  assert_exact_output '', all_stdout
+  expect(last_command_started).to have_output_on_stdout('')
 end
 
 Then /^stdout includes "(.*)"$/ do |text|
-  assert_partial_output(text, all_stdout)
+  expect(last_command_started).to have_output(/#{Regexp.escape(text)}/)
 end
 
 Then /^it succeeds$/ do
-  assert_exit_status Reek::CLI::Application::STATUS_SUCCESS
+  success = Reek::CLI::Application::STATUS_SUCCESS
+  expect(last_command_started).to have_exit_status(success)
 end
 
 Then /^the exit status indicates an error$/ do
-  assert_exit_status Reek::CLI::Application::STATUS_ERROR
+  error = Reek::CLI::Application::STATUS_ERROR
+  expect(last_command_started).to have_exit_status(error)
 end
 
 Then /^the exit status indicates smells$/ do
-  assert_exit_status Reek::CLI::Application::STATUS_SMELLS
+  smells = Reek::CLI::Application::STATUS_SMELLS
+  expect(last_command_started).to have_exit_status(smells)
 end
 
 Then /^it reports:$/ do |report|
-  assert_exact_output "#{report}\n", all_stdout
+  expect(last_command_started).to have_output(report.gsub('\n', "\n"))
```

Comment:
You should use `have_output_on_stdout` here. That way, you don't have to add the extra line `!!! reek has found smells - exiting!` to the scenario's.